grr/server/grr_response_server/gui/ui/components/client/client.ts

```diff
@@ -46,6 +51,36 @@ export class Client implements OnInit, OnDestroy {
         });
   }
 
+  ngAfterViewInit() {
+    this.clientDetailsDrawers.closedStart.subscribe(() => {
+      const urlTokens = this.location.path().split('/');
+      this.location.go(urlTokens.slice(0, -1).join('/'));
+    });
+
+    this.location.onUrlChange(url => {
+      const urlTokens = url.split('/');
+      if (urlTokens[urlTokens.length - 1] === 'details') {
```

Comment:
Please introduce and re-use a function that receives url segments, checks for 'details' and opens the drawer. This function can be used in `onUrlChange` and the following `route.url.pipe`.

Reply:
This will require to make a global function, because in the next PR the logic is split between the `client` component and the `client-overview` component. Is that OK?

Reply:
No, let's keep it separate functions for now then.